include/tvm/script/ir_builder/tir/ir.h

```diff
@@ -430,14 +430,26 @@ void Evaluate(PrimExpr value);
  * \brief Create a TIR var that represents a pointer
  * \param dtype The data type of the pointer.
  * \param storage_scope The storage scope of the pointer.
+ * \param size_var Whether the pointer is a size var.
  * \return The pointer.
  */
-Var Handle(runtime::DataType dtype = runtime::DataType::Void(), String storage_scope = "global");
-
-#define TVM_TIR_IR_BUILDER_DEF_DTYPE_CAST(FuncName, DType)                             \
-  inline PrimExpr FuncName(Optional<PrimExpr> expr = NullOpt) {                        \
-    DataType dtype = DType;                                                            \
-    return expr.defined() ? tvm::cast(dtype, expr.value()) : tvm::tir::Var("", dtype); \
+inline Var Handle(runtime::DataType dtype = runtime::DataType::Void(),  //
+                  String storage_scope = "global",                      //
+                  bool size_var = false) {
```

Comment:
Would be great if we can rename it to `is_size_var`

Reply:
It's a matter of naming, and I personally believe `size_var` is a more proper name for the following reasons:
- It's shorter
- It already implies the semantics of "is_"
- it's consistent with TIR's SizeVar.

Well, I don't hold strong opinion though, so open to change if you insist `is_size_var` be slightly better :-)

Reply:
Updated to `is_size_var` per discussion with @Hzfengsy